src/SourceBuild/content/build.sh

```diff
@@ -193,49 +203,57 @@ if [[ "$ci" == true ]]; then
   if [[ "$exclude_ci_binary_log" == false ]]; then
     binary_log=true
   fi
+
+  properties="$properties /p:ContinuousIntegrationBuild=true"
 fi
 
 . "$scriptroot/eng/common/tools.sh"
 
 function Build {
-  if [[ "$sourceOnly" != "true" ]]; then
+  if [[ "$sourceOnly" == "true" ]]; then
+    "$CLI_ROOT/dotnet" build-server shutdown
+
+    InvokeMsBuild "$scriptroot/eng/tools/init-build.proj" "ExtractToolPackage,BuildMSBuildSdkResolver" "BuildMSBuildSdkResolver"
+
+    # kill off the MSBuild server so that on future invocations we pick up our custom SDK Resolver
+    "$CLI_ROOT/dotnet" build-server shutdown
 
+    # Point MSBuild to the custom SDK resolvers folder, so it will pick up our custom SDK Resolver
+    export MSBUILDADDITIONALSDKRESOLVERSFOLDER="$scriptroot/artifacts/toolset/VSSdkResolvers/"
+    
+  fi
+
+  InvokeMsBuild "$scriptroot/build.proj" "Build" "Build"
+}
+
+function Test {
+  InvokeMsBuild "$scriptroot/build.proj" "Test" "Test"
+}
+
+function InvokeMsBuild {
+  local projectPath="$1"
+  local target="$2"
+  local logName="$3"
+
+  if [[ "$sourceOnly" == "true" ]]; then
+    "$CLI_ROOT/dotnet" msbuild "$projectPath" \
+      -t:$target \
+      -bl:"$scriptroot/artifacts/log/$configuration/$logName.binlog" \
+      -flp:LogFile="$scriptroot/artifacts/log/$configuration/$logName.log" \
+      -clp:v=m \
+      $properties
+  else
     InitializeToolset
 
     local bl=""
     if [[ "$binary_log" == true ]]; then
-      bl="/bl:\"$log_dir/Build.binlog\""
+      bl="/bl:\"$log_dir/$logName.binlog\""
     fi
-
-    MSBuild "$scriptroot/build.proj" \
+    MSBuild "$projectPath" \
+      -t:$target \
       $bl \
       /p:Configuration=$configuration \
```

Comment:
Can Configuration be added as part of `$properties` rather than passed separately?

Reply:
In the interest of limiting the changes in this PR, I'm going to decline making this change since this implementation already existed.